### PR TITLE
feat(wallet-ui): provision new wallet

### LINKED ISSUE #6211: UI for getting a smart wallet
## What is the Problem Being Solved?

UI for making a `provision-one` transaction.

## Description of the Design

?

(I can provide some references on back-end stuff)

## Test Plan

cc @samsiegart @rowgraus

### PR DESCRIPTION
**Pull request by samsiegart**
fixes: https://github.com/Agoric/agoric-sdk/issues/6211

### CODE REVIEW

File: packages/wallet/ui/src/components/ProvisionDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  const provisionWallet = async () => {
+    const {
+      signers: { interactiveSigner },
+    } = keplrConnection;
+    if (!interactiveSigner) {
+      setError(errors.NO_SIGNER);
+      return;
+    }
+
+    setError('');
```
Comment (turadg) [inline packages/wallet/ui/src/components/ProvisionDialog.jsx]: why is this necessary? seems an odd place for it. do you want to reset the error when the user initiates a new action?

why use empty string instead of null or undefined?

File: packages/wallet/ui/src/components/ProvisionDialog.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+              <b>Network Config</b>:{' '}
+              <Link href={href} underline="none" color="rgb(0, 176, 255)">
+                {href}
+              </Link>
+            </DialogContentText>
+            <DialogContentText sx={{ pt: 2 }}>
+              <b>Wallet Address:</b> {address}
+            </DialogContentText>
+            <DialogContentText sx={{ pt: 2 }}>
+              There is no smart wallet provisioned for this address yet. A fee
```
Comment (turadg) [inline packages/wallet/ui/src/components/ProvisionDialog.jsx:87]: if we're not going to read it from params, at least make it a const. should have a comment indicating the tech debt and why

File: packages/wallet/ui/src/components/SmartWalletConnection.jsx [hunk 3]
```diff
@@ -45,11 +54,32 @@ const SmartWalletConnection = ({
...
+  const { href, smartConnectionMethod } = connectionConfig;
+  let publicAddress;
+  if (
+    smartConnectionMethod === SmartConnectionMethod.KEPLR &&
+    keplrConnection
+  ) {
+    publicAddress = keplrConnection.address;
+  } else if (smartConnectionMethod === SmartConnectionMethod.READ_ONLY) {
+    publicAddress = connectionConfig.publicAddress;
+  }
```
Comment (turadg) [inline packages/wallet/ui/src/components/SmartWalletConnection.jsx]: consider a helper method so this is one line here assigning to a const

File: packages/wallet/ui/src/contexts/Provider.jsx [hunk 4]
```diff
@@ -153,7 +152,9 @@ const Provider = ({ children }) => {
     restoredConnectionConfig !== null,
   );
   const [connectionStatus, setConnectionStatus] = useState(
-    wantConnection ? 'connecting' : 'disconnected',
```
Comment (turadg) [inline packages/wallet/ui/src/contexts/Provider.jsx:156]: 👍

File: packages/wallet/ui/src/tests/App.test.jsx [hunk 5]
```diff
@@ -28,7 +28,6 @@ const withApplicationContext =
         allWalletConnections={[
           'http://unit-test-net.agoric.com/network-config',
         ]}
-        connectionState="Connecting"
```
Comment (turadg) [inline packages/wallet/ui/src/tests/App.test.jsx:31]: 👍 one kind of status

File: packages/wallet/ui/src/util/keyManagement.js [hunk 6]
```diff
@@ -391,6 +412,43 @@ export const makeInteractiveSigner = async (
...
+
+      const b64address = toBase64(toAccAddress(address));
+
+      const act1 = {
+        typeUrl: SwingsetMsgs.MsgProvision.typeUrl,
+        /** @type {Provision} */
+        value: {
+          address: b64address,
+          nickname: 'my wallet',
+          // XXX dry with PowerFlags const
```
Comment (turadg) [inline packages/wallet/ui/src/util/keyManagement.js]: how about now?

**Review by turadg — COMMENTED** [review]
Doesn't seem ready for merge. I left some comments. Also the git commits are noisy. Please clean those up for the changelog

**Comment by samsiegart** [comment]
> Also the git commits are noisy. Please clean those up for the changelog

You mean just squash? I'm not sure how else.

File: packages/wallet/ui/src/util/keyManagement.js (hunk 6 above)
Comment (samsiegart) [reply packages/wallet/ui/src/util/keyManagement.js]: One concern I have is when we import these packages just to use a const or type etc, we increase the bundle size by a lot. Maybe it's better to just leave it?

File: packages/wallet/ui/src/components/ProvisionDialog.jsx (hunk 1 above)
Comment (samsiegart) [reply packages/wallet/ui/src/components/ProvisionDialog.jsx]: Yea we want to reset the error when the user initiates a new action.

I sometimes get in the habit of using empty string because certain input elements don't play well with null, but no reason otherwise. Typically I never set things to `undefined` explicitly but maybe that's okay as well.

File: packages/wallet/ui/src/components/ProvisionDialog.jsx (hunk 2 above)
Comment (samsiegart) [reply packages/wallet/ui/src/components/ProvisionDialog.jsx:87]: Made a const and TODO

File: packages/wallet/ui/src/components/SmartWalletConnection.jsx (hunk 3 above)
Comment (samsiegart) [reply packages/wallet/ui/src/components/SmartWalletConnection.jsx]: Done

File: packages/wallet/ui/src/components/ProvisionDialog.jsx (hunk 1 above)
Comment (turadg) [reply packages/wallet/ui/src/components/ProvisionDialog.jsx]: > Yea we want to reset the error when the user initiates a new action.

Then please put it where they initiate the action. Either at the top of this function or sooner.

> certain input elements don't play well with null, but no reason otherwise

Please use `null` for "no error". Empty string could be an error that just doesn't have any description. If there are elements where this causes problems I'd like to fix them

File: packages/wallet/ui/src/util/keyManagement.js (hunk 6 above)
Comment (turadg) [reply packages/wallet/ui/src/util/keyManagement.js]: Doesn't the bundler have tree shaking? If not please make a ticket to get that working and include it in the comment here as to why these aren't imported yet

**Comment by turadg** [comment]
> You mean just squash? 

I was thinking an interactive rebase with renaming. E.g. `git rebase --interactive master` or a GUI like Github Desktop. The latter lets you drag commits together to squash particular ones.

File: packages/wallet/ui/src/components/ProvisionDialog.jsx (hunk 1 above)
Comment (samsiegart) [reply packages/wallet/ui/src/components/ProvisionDialog.jsx]: Changed to use null instead of empty string

I didn't want to have two setErrors in the case of the early return above. I just split this into another function to make it more readable if that's any better?

File: packages/wallet/ui/src/util/keyManagement.js (hunk 6 above)
Comment (samsiegart) [reply packages/wallet/ui/src/util/keyManagement.js]: Looks like it does have tree shaking yes. Importing this only increased the bundle size by ~4kB. Still the bundle size is getting up there, so I created a separate issue for that https://github.com/Agoric/agoric-sdk/issues/6291

**Comment by samsiegart** [comment]
> I was thinking an interactive rebase with renaming. E.g. `git rebase --interactive master` or a GUI like Github Desktop. The latter lets you drag commits together to squash particular ones.

Ah okay, yea just tried to clean up the messages now

**Review by turadg — APPROVED** [review]
Approved assuming the PR will be squashed with a `feat` title.
